sync/fromDisk/questions.js

```diff
@@ -56,11 +56,15 @@ module.exports = {
                             external_grading_files = opts.serverFilesCourse;
                         }
                     }
+                    if ((q.type == 'v3') && (q.partialCredit != null)) {
+                        return callback(new Error(`Question ${qid} has type "v3" and so cannot specify "partialCredit" (it automatically uses partial credit)`));
+                    }
                     var params = {
                         uuid: q.uuid,
                         qid: qid,
                         type: (q.type == 'v3') ? 'Freeform' : q.type,
                         title: q.title,
+                        partial_credit: (q.type == 'v3') ? true : !!q.partialCredit,
```

Comment:
This sort of expression always confuses me. Perhaps you could preface it with a comment like this:

`// The expression '!!null' evaluates to false, so the default for non-v3 questions (i.e., the value 
of partial_credit when partialCredit is undefined) is false`